sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -4331,6 +4378,7 @@ public PDone expand(PBegin input) {
         PCollection<String> result =
             input
                 .apply(inputPTransform)
+                .setIsBoundedInternal(isStreaming ? IsBounded.UNBOUNDED : IsBounded.BOUNDED)
```

Comment:
Another way to do this that might be better is to use `TestStream` in the unbounded case. This will probably give best coverage. Even for an unbounded PCollection the watermark might instantly move to infinity.